src/components/Leaderboard.vue

```diff
@@ -25,69 +28,58 @@ const {
-} = useConfirmDialog()
-
-async function confirmDeleteBoard() {
-	// TODO: Error-handling
-	await leaderboards.deleteLeaderboard(props.id, useAuth(token.value))
-	execute()
+async function revealDelete() {
+	if (confirm('Really delete this leaderboard? (This action can be reversed)')) {
+		try {
+			await leaderboards.deleteLeaderboard(props.id, useAuth(token.value))
+			execute()
+		} catch (error: unknown) {
+			updateError.value = 'Failed to delete: ' + (error as Response).status.toString(10)
+		}
+	}
 }
 
-async function confirmRestoreBoard() {
-	// TODO: Error-handling
-	await leaderboards.restoreLeaderboard(props.id, useAuth(token.value))
-	execute()
+async function revealRestore() {
+	if (confirm('Really restore this leaderboard? (This action can be reversed)')) {
+		try {
+			await leaderboards.restoreLeaderboard(props.id, useAuth(token.value))
+			execute()
+		} catch (error: unknown) {
+			updateError.value = 'Failed to restore: ' + (error as Response).status.toString(10)
+		}
+	}
 }
 </script>
 
 <template>
 	<div class="container">
 		<div v-if="isLoading">Loading...</div>
-		<div v-else-if="error" class="error-container">
-			<p>Error</p>
+		<div v-else-if="error" class="errorContainer">
+			<RouterLink class="backLink" :to="{ name: 'leaderboardsList' }">&lt; Back</RouterLink>
+			<p class="errorText">{{ error }}</p>
 			<button @click="execute()" class="button">Reload</button>
-			<!-- TODO: Add button to go back to the List page -->
 		</div>
 
-		<div v-else class="main-content">
-			<!-- Deletion confirmation -->
-			<div v-if="isRevealedDelete" class="delete-confirmation-container">
-				<button @click="cancelDelete">Cancel</button>
-				<button @click="confirmDeleteBoard">Confirm</button>
-			</div>
-
-			<!-- Restoration confirmation -->
-			<div v-if="isRevealedRestore" class="delete-confirmation-container">
-				<button @click="cancelRestore">Cancel</button>
-				<button @click="confirmRestoreBoard">Confirm</button>
-			</div>
-
-			<RouterLink class="back-link" :to="{ name: 'leaderboardsList' }">&lt; Back</RouterLink>
-			<div class="action-button-container">
+		<div v-else class="mainContent">
+			<RouterLink class="backLink" :to="{ name: 'leaderboardsList' }">&lt; Back</RouterLink>
+			<div class="actionButtonContainer">
 				<!-- TODO: Create Edit page, and then add this link to it -->
-				<!-- <RouterLink to="/edit"><button class="action-button">✎</button></RouterLink> -->
+				<!-- <RouterLink to="/edit"><button class="actionButton">✎</button></RouterLink> -->
 				<button
 					v-if="board?.deletedAt === null"
-					class="action-button delete-button"
+					class="actionButton deleteButton"
 					@click="revealDelete"
 				>
 					Delete
 				</button>
-				<button v-else class="action-button" @click="revealRestore">
+				<button v-else class="actionButton" @click="revealRestore">
 					Restore
 				</button>
 			</div>
 
+			<span v-if="updateError !== null" class="errorText">{{ updateError }}</span>
```

Comment:
Change this to `v-if="updateError"` because empty string is falsey. Also consider using `<p>` instead.